src/test/java/com/microsoft/sqlserver/jdbc/unit/lobs/LobsStreamingTest.java

```diff
@@ -0,0 +1,268 @@
+        tableName = RandomUtil.getIdentifier("streamingTest");
+    }
+
+    private String getRandomString(int length, String validCharacters) {
+        StringBuilder salt = new StringBuilder();
+        Random rnd = new Random();
+        while (salt.length() < length) {
+            int index = (int) (rnd.nextFloat() * validCharacters.length());
+            salt.append(validCharacters.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+    }
+
+    // closing the scanner closes the Inputstream, and the driver needs the stream to fill LoBs
+    @SuppressWarnings("resource")
+    private String getStringFromInputStream(InputStream is) {
+        java.util.Scanner s = new java.util.Scanner(is, java.nio.charset.StandardCharsets.US_ASCII).useDelimiter("\\A");
+        return s.hasNext() ? s.next() : "";
+    }
+
+    private String getStringFromReader(Reader r, long l) throws IOException {
+        // read the Reader contents into a buffer and return the complete string
+        final StringBuilder stringBuilder = new StringBuilder((int) l);
+        char[] buffer = new char[(int) l];
+        while (true) {
+            int amountRead = r.read(buffer, 0, (int) l);
+            if (amountRead == -1) {
+                break;
+            }
+            stringBuilder.append(buffer, 0, amountRead);
+        }
+        return stringBuilder.toString();
+    }
+
+    private void createLobTable(Statement stmt, String table, LoB l) throws SQLException {
+        String columnType = (l == LoB.CLOB) ? "varchar(max)" : "nvarchar(max)";
+        stmt.execute("CREATE TABLE [" + table + "] (id int, lobValue " + columnType + ")");
+    }
+
+    private ArrayList<String> createRandomStringArray(LoB l) {
+        String characterPool = (l == LoB.CLOB) ? ASCII_CHARACTERS : UNICODE_CHARACTERS;
+        ArrayList<String> string_array = new ArrayList<>();
+        IntStream.range(0, LOB_ARRAY_SIZE).forEach(i -> string_array.add(
+                getRandomString(ThreadLocalRandom.current().nextInt(LOB_LENGTH_MIN, LOB_LENGTH_MAX), characterPool)));
+        return string_array;
+    }
+
+    private void insertData(Connection conn, String table, ArrayList<String> lobs) throws SQLException {
+        try (PreparedStatement pstmt = conn.prepareStatement("INSERT INTO [" + table + "] VALUES(?,?)")) {
+            for (int i = 0; i < lobs.size(); i++) {
+                Clob c = conn.createClob();
+                c.setString(1, lobs.get(i));
+                pstmt.setInt(1, i);
+                pstmt.setClob(2, c);
+                pstmt.addBatch();
+            }
+            pstmt.executeBatch();
+        }
+    }
+
+    @Test
+    @DisplayName("testLengthAfterStream")
+    public void testLengthAfterStream() throws SQLException, IOException {
+        try (Connection conn = DriverManager.getConnection(connectionString);) {
+            try (Statement stmt = conn.createStatement()) {

```

Comment:
again nitpicking...why not put that in the same try block and save an indentation level?